### PR TITLE
ENP-3161: Report failed index in repeated fields

### PR DESCRIPTION
**Pull request by lachlancooper**
## Summary

When validation fails on a repeated field, report the index of the specific element that failed.

## Problem description

Currently when a repeated field fails validation, it can be hard to determine which specific element was to blame.

## Pros/cons of approach implemented

Pro: Reporting only the index avoids logging any significant input information, while still allowing the requester to easily determine which element is at fault.

Con: This only reports the index of the _first_ failed element. This could require the requester to make multiple requests to find all failed elements.

## Checklist

- [x] Is this PR a reasonable size?

- [ ] List deployment sequence with all relevant PRs.
  - ...

---

**Code Review Guidelines** for Reviewers

- Try to review in a timely manner. Opinions/nitpicks should not be blockers. Pair on a call for non-trivial feedback.
- Overall design and approach should follow established patterns. Don't try to make the PR perfect.
- Try to identify edge cases, race conditions, over-engineering, lack of test coverage and complexity.
- If you don't feel qualified to review the code, pass it on to someone who is.

### CODE REVIEW

File: protobuf/protoc-gen-govalidator/plugin/plugin.go [hunk 1]
```diff
@@ -913,13 +913,16 @@ func genMsgValidator(g *protogen.GeneratedFile, f *protogen.Field, varName strin
...
-		g.P("for _, item := range ", varName, "{")
+		g.P("for i, item := range ", varName, "{")
 		varName = "item"
 	}
 
 	g.P("if ", varName, " != nil {")
 	g.P("if v, ok := interface{}(", varName, ").(", s12protoPackage.Ident("Validator"), "); ok {")
 	g.P("if err := v.Validate(); err != nil {")
+	if repeated {
+		g.P("err = ", fmtPackage.Ident("Errorf"), "(\"at index %d: %v\", i, err)")
```
Comment (sc-peter) [inline protobuf/protoc-gen-govalidator/plugin/plugin.go:924]: Thanks for adding this @lachlancooper. Did you check how this works with multiple validation errors in different messages? We should also add tests for this to make sure that it's working as expected. I can take a look later in the week if you'd like.

Comment (lachlancooper) [reply protobuf/protoc-gen-govalidator/plugin/plugin.go:924]: Good question, I guess currently validation errors don't report the specific message that failed?

I agree about adding tests, will move this to draft until we get a chance to address that.
